src/main/resources/com/puppycrawl/tools/checkstyle/packages_1_1.dtd

```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!ELEMENT checkstyle-packages (package)*>
+
+<!ELEMENT package (package)*>
+<!ATTLIST package name NMTOKEN #REQUIRED>
```

Comment:
@romani 
> We enforce all files to have newline at the end

No, we do not. See https://github.com/MEZk/checkstyle/blob/52d1237a32dd1ddd6ea4697643534f59d8bebb64/config/checkstyle_checks.xml#L21-L21 .

[The documentation](https://github.com/MEZk/checkstyle/blob/131723cab4ac2fd0953b877515a59399c81de002/src/xdocs/config_misc.xml#L1191-L1191) of NewLineAtEndOfFile is not correct. If we do not set fileExtensions property of AbstractFileSetCheck (NewLineAtEndOfFile extends AbstractFileSetCheck) Checker uses its own property fileExtensions.